src/runtime/auto-prompt-manager.js

```diff
@@ -502,26 +502,40 @@ export class AutoPromptManager {
       autoPromptType === AutoPromptType.CONTRIBUTION ||
       autoPromptType === AutoPromptType.CONTRIBUTION_LARGE
     ) {
-      const preferSurveyOverContributionPrompt =
-        await this.isExperimentEnabled_(
-          article,
-          ExperimentFlags.SURVEY_TRIGGERING_PRIORITY
+      // Survery take highest priority if this flag is enabled.
+      const prioritizeSurvey = await this.isExperimentEnabled_(
+        article,
+        ExperimentFlags.SURVEY_TRIGGERING_PRIORITY
+      );
+
+      let previousPrompts = [];
```

Comment:
Nit: This name could be clearer
```suggestion
      let previouslyShownPrompts = [];
```

Reply:
Done.